app/controllers/developers_controller.rb

```diff
@@ -44,12 +46,24 @@ def filter
 private
 
   def sort
-    { 
+    {
       "Rate: Ascending" => lambda { |active| active.order(rate: :asc) },
       "Rate: Descending" => lambda { |active| active.order(rate: :desc) }
     }
   end
 
+  def selection_filter(initial_developers)
+    preload = initial_developers.joins(:tags)
+    intersection_query = selected_tags.reduce(initial_developers) do |acc, tag_name|
+      acc.merge(preload.where(tags: {name: tag_name}))
+    end
+  end
+
+  def selected_tags
+    params["filter"] ||= {}
+    params["filter"].select{ |key, val| val == "1" }.keys
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.<br>Space missing to the left of {.